### PR TITLE
P0 [dev.net] - Allow hostnames to be parsed for Narwhal config

### PR DESCRIPTION
**Pull request by velvia**
Quick review - allow hostnames to be used for Narwhal and Genesis/network configuration, not just IP addresses.

This is needed for AWS dev.net deployment (using local domain names, like in K8s, instead of fixed IPs)

### CODE REVIEW

**Comment by bmwill** [comment]
looking at this I'm fairly confused on why this would be needed. At the end of the day the exact same code path is going to be called (if you look though the stdlib). this did work at some point so maybe there's something misconfigured in the environment you're running in?

**Comment by velvia** [comment]
@bmwill sorry the code was wrong, hopefully the new code will make it a bit more clear.   The `to_socket_addrs()` function returns an Iter of addresses, it's not at all the same as the old `parse()` method.  

`parse()` does not work with hostnames.  It gives the same error as I got in a container:

```
thread 'rest_server_tests::test_concurrency' panicked at 'called `Result::unwrap()` on an `Err` value: AddrParseError(())', sui/src/config.rs:106:49
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I have changed the default hostname to `localhost` which did not work before in tests.

File: sui/src/config.rs [hunk 1]
```diff
@@ -69,6 +70,15 @@ impl Clone for AuthorityPrivateInfo {
     }
 }
 
+fn socket_addr_from_hostport(host: &str, port: u16) -> SocketAddr {
+    let mut addresses = format!("{host}:{port}")
+        .to_socket_addrs()
```
Comment (mystenmark) [inline sui/src/config.rs:75]: i'm probably being too paranoid, but adding panicking and blocking functions for things that might actually fail (e.g. dns resolution) is scary. i know its only being called at start up, but you could imagine someone moving this to some other location and using it unsafely.

a short warning comment would make me happy 😄

Comment (velvia) [reply sui/src/config.rs:76]: @mystenmark ok comment added, thanks

**Review by bmwill — APPROVED** [review]

**Review by mystenmark — APPROVED** [review]

**Pull request merged**